app/controllers/api/generic_object_definitions_controller.rb

```diff
@@ -18,13 +17,78 @@ def edit_resource(type, id, data)
     end
 
     def delete_resource(type, id, data = {})
-      id ||= data['name']
-      go_def = resource_search(id, type, collection_class(:generic_object_definitions))
+      go_def = fetch_generic_object_definition(type, id, data)
       go_def.destroy!
     rescue => err
       raise BadRequestError, "Failed to delete generic object definition - #{err}"
     end
 
+    def add_attributes_resource(type, id, data)
+      go_def = fetch_generic_object_definition(type, id, data)
+      attributes = data['attributes'] || data['resource']['attributes']
```

Comment:
isn't this handled by the base class, ie. if data['resource'] exists, that will be the data coming in ?

Reply:
@abellotti not in the case where the id is the name of the generic object

Reply:
👍